examples/rotation.rs

```diff
@@ -14,174 +14,170 @@ use wlroots::utils::{init_logging, WLR_DEBUG};
 use wlroots::wlroots_sys::wl_output_transform;
 use wlroots::xkbcommon::xkb::keysyms;
 
-const CAT_WIDTH: u32 = 128;
-const CAT_HEIGHT: u32 = 128;
-const CAT_DATA: &'static [u8] = include_bytes!("cat.data");
-
-/// Helper step by iterator, because `step_by` on `Range` is unstable.
-struct StepRange(i32, i32, i32);
-impl Iterator for StepRange {
-    type Item = i32;
-    fn next(&mut self) -> Option<i32> {
-        if self.0 < self.1 {
-            let v = self.0;
-            self.0 = v + self.2;
-            Some(v)
-        } else {
-            None
-        }
+const CAT_TEXTURE_WIDTH: u32 = 128;
+const CAT_TEXTURE_HEIGHT: u32 = 128;
+const CAT_TEXTURE_DATA: &'static [u8] = include_bytes!("cat.data");
+const VELOCITY_STEP_DIFF: f32 = 16.0;
+
+struct Vector2 {
```

Comment:
Don't make a named struct, please use a tuple instead directly in the state struct.